[MIG][15.0] product: migration for 15.0

daonamutc1 opened the pull request:
[Task](https://viindoo.com/web#id=23069&action=409&active_id=177&model=project.task&view_type=form&cids=1&menu_id=289)
- Rename filed product_template.type to product_template.detailed_type
- Compute value_count of product.template.attribute.line
- Convert text to html

Video:
- Before: https://drive.google.com/file/d/1aLD1Udmq0jt8kcr9iITmSPyASKWoDzQ6/view?usp=sharing
- After: https://drive.google.com/file/d/13TJJdBGbV5nuwVjAFjmNlhZq3Eurpu1i/view?usp=sharing

tranngocson1996 commented on openupgrade_scripts/scripts/product/15.0.1.2/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,55 @@
+product      / product.template         / type (selection)              : now a function
+# NOTHING TO DO: new fields
+
+product      / product.template.attribute.line / value_count (integer)         : NEW isfunction: function, stored
```
@daonamutc1 chỗ này anh khả năng sẽ phải tạo cột trước rồi fill dữ liệu vào trước cho nó. Tránh compute gây chậm do nhiều record

royle-vietnam commented on openupgrade_scripts/scripts/product/15.0.1.2/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,57 @@
+product      / product.attribute.value  / color (integer)               : NEW hasdefault
+product      / product.packaging        / _order                        : _order is now 'product_id, sequence, id' ('sequence')
+product      / product.product          / _order                        : _order is now 'priority desc, default_code, name, id' ('default_code, name, id')
+product      / product.product          / description (text)            : type is now 'html' ('text')
```
@daonamutc1 cần dùng hàm `convert_field_to_html` để chuyển giá trị `text` thành `html` anh nhé

royle-vietnam commented on openupgrade_scripts/scripts/product/15.0.1.2/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,57 @@
+product      / product.product          / description (text)            : type is now 'html' ('text')
+product      / product.product          / product_template_variant_value_ids (many2many): NEW relation: product.template.attribute.value
+product      / product.template         / _order                        : _order is now 'priority desc, name' ('name')
+product      / product.template         / description (text)            : type is now 'html' ('text')
```
@daonamutc1 tương tự như trên

daonamutc1 committed:
[MIG] product: migration for 15.0
- Compute filed product_template.type and product_template.detailed_type
- Compute value_count
- Convert text to html
